ecosystem/typescript/sdk_v2/src/internal/account.ts

```diff
@@ -121,3 +122,19 @@ export async function getResource(args: {
   });
   return data;
 }
+
+export async function estimateAccountMaxGasAmount(args: { aptosConfig: AptosConfig; accountAddress: HexInput }) {
+  const { aptosConfig, accountAddress } = args;
+  // Only APT coin is accepted as gas
+  const typeTag = `0x1::coin::CoinStore<${APTOS_COIN}>`;
+  const [{ gas_estimate: gasUnitPrice }, resources] = await Promise.all([
+    getGasPriceEstimation({ aptosConfig }),
+    getResources({ aptosConfig, accountAddress }),
```

Comment:
We should use `getResource()`, rather than `getResources()`, since we know the resource.  Otherwise, if you have > 10k resources, you'll never find it and it'll be expensive.